### PR TITLE
Segment email send to email address just once

### PR DESCRIPTION
**Pull request by kuzmany**
| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | "features" for all features, enhancements and bug fixes (until 3.3.0 is released) 
| Bug fix?                               | yes/no
| New feature?                           | yes/no
| Deprecations?                          | yes/no
| BC breaks?                             | yes/no
| Automated tests included?              | yes/no
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:

Our client want use contacts with same email address, but avoid send more times to that email address, even If contact's are in segments. For example If more employees use same email address, or family use one email address for parents and children.
This PR added new configuration parameter to enable avoid send segment email to more like one unique email address.

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Create segment
3. Create 3 contacts with same  email@domain.tld and add them to segment
4. Create segment email
4. Go to configuration and enable new options - Send to duplicated just once 
image
5. Send segment email
6. See just one email you receive

5. 

<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/9781"></a>

### CODE REVIEW

**Review by volha-pivavarchyk — DISMISSED** [review]
Works well as expected

**Comment by jos0405** [comment]
@kuzmany could you plz update this? I'll test immediately and it could be merged.

**Comment by kuzmany** [comment]
@jos0405 done

**Review by jos0405 — DISMISSED** [review]
Tested, works fine!

**Review by volha-pivavarchyk — DISMISSED** [review]
Works well

File: app/bundles/EmailBundle/Entity/EmailRepository.php [hunk 1]
```diff
@@ -15,6 +15,11 @@
  */
 class EmailRepository extends CommonRepository
 {
+    /**
+     * @var bool
+     */
+    protected $segmentEmailOnceToEmailAddress;
```
Comment (escopecz) [inline app/bundles/EmailBundle/Entity/EmailRepository.php]:
```suggestion
    private bool $segmentEmailOnceToEmailAddress = false;
```

File: app/bundles/EmailBundle/Tests/Model/EmailModelFunctionalTest.php [hunk 2]
```diff
@@ -47,4 +49,67 @@ public function testNotOverwriteChildrenTranslationEmailAfterSaveParent(): void
...
+        $this->em->persist($email2);
+
+        /** @var EmailModel $emailModel */
+        $emailModel = self::$container->get('mautic.email.model.email');
+
+        [$sentCount, $failedCount, $failedRecipientsByList] = $emailModel->sendEmailToLists($email);
+
+        self::assertEquals(2, $sentCount, $email->getCustomHtml());
+
+        $this->connection->executeQuery("SET GLOBAL sql_mode=(SELECT REPLACE(@@sql_mode,'ONLY_FULL_GROUP_BY',''))");
```
Comment (escopecz) [inline app/bundles/EmailBundle/Tests/Model/EmailModelFunctionalTest.php:105]: Why is this here? I think that's why the transaction rollback fails.

**Review by escopecz — CHANGES_REQUESTED** [review]
There is a failing test:
```
There was 1 error:

1) Mautic\EmailBundle\Tests\Model\EmailModelFunctionalTest::testSendEmailToTwoSameEmailAddressWithOptionSegmentEmailOnceToEmailAddress
LogicException: You cannot re-create the client when a transaction rollback for cleanup is enabled. Turn it off using $useCleanupRollback property or avoid re-creating a client.

/home/runner/work/mautic/mautic/app/bundles/CoreBundle/Test/MauticMysqlTestCase.php:85
/home/runner/work/mautic/mautic/app/bundles/EmailBundle/Tests/Model/EmailModelFunctionalTest.php:106
phpvfscomposer:///home/runner/work/mautic/mautic/vendor/phpunit/phpunit/phpunit:97
```

**Comment by npracht** [comment]
Closed in favor of https://github.com/mautic/mautic/pull/13768 rebased against M5.

**Pull request closed without merging**